ibis/expr/types/relations.py

```diff
@@ -465,49 +468,55 @@ def sort_by(
 
     def union(
         self,
-        right: Table,
+        *tables: Table,
```

Comment:
How about `right, *tables` for all of these to ensure that at least one table is present? This would also ensure backwards compatibility.

Reply:
> a bit of a weird signature to the user

I would argue it's also weird that t.union(*args) may return `t`.

> how do I refer to the tables being joined

`self`, `right`, and `*tables` seems fine to me

> Do we care that a positional argument has been renamed?

I personally don't, but if it's straightforward to keep it backwards compatible then it seems like we should do it.

Reply:
> `self`, `right`, and `*tables` seems fine to me

That still seems awkward to write out when really a top-level function here would want `*tables` alone with a requirement of 2 or more tables. `right` and `*tables` (and self for `union`/`intersect`) are identical in purpose, but calling the args different things makes the user question they have different meanings.

How would you document `union` with that signature?

> I would argue it's also weird that `t.union(*args)` may return `t`.

Good catch, we shouldn't allow that. Technically it'd be fine if `distinct=False`, but `ibis.union(t, distinct=True)` should drop duplicates from `t`, and calling out to a different way of handling that within a `union` call would be weird.